src/main/java/org/terasology/module/health/systems/RegenAuthoritySystem.java

```diff
@@ -111,6 +98,8 @@ private void invokeRegenOperations(long currentWorldTime) {
                 .forEach(regenEntity -> {
                     RegenComponent regen = regenEntity.getComponent(RegenComponent.class);
                     regenSortedByTime.remove(regen.soonestEndTime, regenEntity);
+                    //TODO: Add a flag to indicate the regeneration id should be removed as soon as the entity is restored to full health?
+                    //      On the other hand, a system can react to the OnFullyHealedEvent and unregister the regeneration id if desired.
```

Comment:
I tend toward removing this piece of logic to remove "completed" regeneration actions, but I'm not sure what the _usual use case_ would be here. I can imagine having a flag to indicate that the regen action is a "only-once" action and should be removed, but I'd like to explore how complicated the solution via `OnFullyHealedEvent` turns out.